### PR TITLE
Editing ucrCore and finishing off Scatterplot

### PR DESCRIPTION
**Pull request by maxwellfundi**
@africanmathsinitiative/developers This is ready for review

Just one comment: There is a developer error that still need fixing

### CODE REVIEW

**Comment by dannyparsons** [comment]
Developer error still occurring? Save control resetting correctly?

**Comment by maxwellfundi** [comment]
@dannyparsons i have mentioned the developer error. The save is now resetting correctly and works well.

**Comment by dannyparsons** [comment]
Oh yeah. I'll check on the developer error.

**Comment by maxwellfundi** [comment]
Okay, great.

**Comment by dannyparsons** [comment]
So when the dialog loads it sets the selector as Nothing. Which means it thinks it hasn't converted to a selector by data frame correctly, because it's also nothing. So we just need to only set the ucrVariableSelector when the selector passed in is not nothing.

**Comment by maxwellfundi** [comment]
@dannyparsons . thats now done

**Comment by lilyclements** [comment]
@maxwellfundi I will give this a thorough review in about 10 minutes.
If there are only minor changes I can see, is it okay if I pull this branch, make the changes, and then either push them back into this branch or push them into the master?

**Comment by dannyparsons** [comment]
@lilyclements On behalf of everyone I think we can agree you don't need to wait for permission to do these changes - feel free to go ahead and make changes on branches. Unless you're not sure if someone is still working on the branch, that is.

**Comment by AlexSananka** [comment]
I too agree on this,especially on the designer changes now that she's the "designer manager".
@lilyclements can just be pulling branches and make the necessary changes,I think this way we can save a lot of time in the review process.

**Comment by lilyclements** [comment]
I do think it's good for me to comment just to say if that's what I'm doing, however. Just in case @dannyparsons is about to merge it, or if someone else decides midway to send a commit

File: instat/dlgScatterPlot.vb [hunk 1]
```diff
@@ -108,14 +113,15 @@ Public Class dlgScatterPlot
...
         ucrFactorOptionalReceiver.SetRCode(clsRaesFunction, bReset)
+        ucrSaveScatterPlot.SetRCode(clsBaseOperator, bReset)
     End Sub
 
     Private Sub TestOkEnabled()
         'tests when okay Is enable. Both x and y aesthetics are mandatory but can be set to x="" or(exclusive) y="" in case the other one is filled. 
-        If (ucrReceiverX.IsEmpty() AndAlso ucrVariablesAsFactorForScatter.IsEmpty) OrElse (ucrSaveScatterPlot.chkSaveGraph.Checked AndAlso ucrSaveScatterPlot.ucrInputGraphName.IsEmpty) Then
-            ucrBase.OKEnabled(False)
-        Else
+        If Not ucrVariablesAsFactorForScatter.IsEmpty AndAlso ucrSaveScatterPlot.IsComplete Then
```
Comment (lilyclements) [inline instat/dlgScatterPlot.vb]: Why did you change the TestOK? Before the change, it allowed you to have either one of the (non-optional) receivers full - as long as at least one of the two are - without any issues (to my knowledge!)

Comment (dannyparsons) [inline instat/dlgScatterPlot.vb]: Good point!

Comment (maxwellfundi) [inline instat/dlgScatterPlot.vb]: I changed that because for this dialog, we just need the y aesthetic mandatory. If all the other receivers are empty then it passes x="".  I feel there is no good point to check if either of other receivers are filled or not.
Why would we need more check to that?

Comment (dannyparsons) [inline instat/dlgScatterPlot.vb]: Try using this dialog in the old version and have y empty.

**Comment by stevekogo** [comment]
What is the state of this?

File: instat/dlgScatterPlot.vb (hunk 1 above)
Comment (maxwellfundi) [inline instat/dlgScatterPlot.vb]: Okay, let me have a look

Comment (maxwellfundi) [inline instat/dlgScatterPlot.vb]: @dannyparsons  i now understand what you mean, if either of y or x is empty, "" is passed in but, both can not be empty

**Comment by maxwellfundi** [comment]
@stevekogo its about to be finished

**Comment by maxwellfundi** [comment]
@lilyclements  i have addressed the issue of test okay. This is now ready for a final review. 

Thanks
